WIP: Update to Python 3 and modern Django

garw opened the pull request:
Update to Python3

garw committed:
Python 3 update: First draft version

This is the first draft version with update to
Python 3, Django 3.1 and the new django registration
module. Basic login and item registration functionality
seems functional.

Known broken:
- Everything RT related
- Probably everything invoice rendering related

garw committed:
Reformat code with black

Reformat the whole codebase with python-black to
obtain a modern python code formatting.

garw committed:
Fix invoice creation

Fix ODT and PDF offer and invoice creation

garw committed:
Fix MEDIA_ROOT usage

Fix usage of MEDIA_ROOT as it is now a pathlib
Path and not a string anymore.

garw committed:
Fix RT functionality

Replace the (with Python 3 still incompatible)
python-rtkit module with rt and adapt all
RT functionality to work again.

garw committed:
Remove old site-local config example

Due to refactoring the old site-local config is now obsolete
the new config is in local_settings.py.example. The settings.py
is now static for all installations and should not be touched
anymore.

Moreover, add the local_settings.py to the example symlink to the
ignorelist, so others can remove this when needed for their
development environment.

towo commented on LICENSE:
```diff
@@ -1,4 +1,8 @@
 The complete software part and website templates is licensed under the MIT License (see below).
```
```suggestion
The software as well as the website templates are licensed under the MIT License (see below).
```

towo commented on LICENSE:
```diff
@@ -1,4 +1,8 @@
 The complete software part and website templates is licensed under the MIT License (see below).
+However, we use the python rt module that is licensed und GPLv3. As a consequence, this whole
+project is distributed under the terms of the GPLv3 even though all SaBot code itself is lisenced
```
```suggestion
project is distributed under the terms of the GPLv3, even though all SaBot code itself is licensed
```

towo commented on LICENSE:
```diff
@@ -1,4 +1,8 @@
 The complete software part and website templates is licensed under the MIT License (see below).
+However, we use the python rt module that is licensed und GPLv3. As a consequence, this whole
+project is distributed under the terms of the GPLv3 even though all SaBot code itself is lisenced
+under the terms of the MIT License. If you want to use it under these terms, either disable the RT
```
```suggestion
under the terms of the MIT License. If you want to use it under those terms, either disable the RT
```

towo commented on devroom/migrations/0001_initial.py:
```diff
@@ -17,36 +17,122 @@ class Migration(migrations.Migration):
+                    "plannedProgram",
+                    models.TextField(
+                        blank=True,
+                        verbose_name="Give a short description of the program you plan for the Devroom. We will choose the projects based on this description. If you have special requirements, please note this also here.",
```
```suggestion
                        verbose_name="Give a short description of the program you plan for the Devroom. We will choose the projects based on this description. If you have special requirements, please note those here as well.",
```

towo commented on devroom/migrations/0001_initial.py:
```diff
@@ -17,36 +17,122 @@ class Migration(migrations.Migration):
+                    "schedule",
+                    models.TextField(
+                        blank=True,
+                        verbose_name="How long do you want to use the dev room? (e.g. one day/both days/only one afternoom) If you don't want the complete time, do you have a preferred day/time?",
```
Should probably try to figure out if the wording should be "dev room",  "devroom" or similar and pick a unified choice.

towo commented on devroom/models.py:
```diff
@@ -2,35 +2,75 @@
+    logo = models.ImageField(
+        blank=True, upload_to="devrooms/logos", verbose_name=_("Project logo")
+    )
+    homepage = models.URLField(blank=True, verbose_name=_("Project homepage url"))
```
```suggestion
    homepage = models.URLField(blank=True, verbose_name=_("Project homepage URL"))
```

towo commented on devroom/models.py:
```diff
@@ -2,35 +2,75 @@
+    plannedProgram = models.TextField(
+        blank=True,
+        verbose_name=_(
+            "Give a short description of the program you plan for the Devroom. We will choose the projects based on this description. If you have special requirements, please note this also here."
```
```suggestion
            "Give a short description of the program you plan for the Devroom. We will choose the projects based on this description. If you have special requirements, please note those here as well."
```

Why the doubling?

towo commented on devroom/views.py:
```diff
@@ -1,26 +1,28 @@
+    def form_valid(self, form):
+        roomId = self.kwargs.get("pk", None)
+        if roomId is None:
+            raise ImproperlyConfigured("You have to provide a pk for the room")
```
`PK` could probably use some clarification…

towo commented on exhibitor/migrations/0001_initial.py:
```diff
@@ -17,36 +17,134 @@ class Migration(migrations.Migration):
+                ),
+                (
+                    "homepage",
+                    models.URLField(blank=True, verbose_name="Project homepage url"),
```
```suggestion
                    models.URLField(blank=True, verbose_name="Project homepage URL"),
```

towo commented on exhibitor/models.py:
```diff
@@ -7,40 +7,78 @@
+    logo = models.ImageField(
+        blank=True, upload_to="exhibitors/logos", verbose_name=_("Project logo")
+    )
+    homepage = models.URLField(blank=True, verbose_name=_("Project homepage url"))
```
```suggestion
    homepage = models.URLField(blank=True, verbose_name=_("Project homepage URL"))
```

towo commented on invoice/migrations/0002_auto_20160207_1240.py:
```diff
@@ -7,20 +7,27 @@
+            model_name="invoice",
+            name="payed",
+            field=models.BooleanField(
+                default=False, verbose_name="This invoice is payed."
```
```suggestion
                default=False, verbose_name="This invoice has been paid."
```

towo commented on invoice/models.py:
```diff
@@ -11,91 +11,145 @@
+    )
+    dueDate = models.DateField(verbose_name=_("Due date"))
+    payed = models.BooleanField(
+        default=False, editable=False, verbose_name=_("This invoice is payed.")
```
```suggestion
        default=False, editable=False, verbose_name=_("This invoice has been paid.")
```

towo commented on invoice/migrations/0003_auto_20160207_1343.py:
```diff
@@ -8,61 +8,105 @@
+            model_name="invoice",
+            name="payed",
+            field=models.BooleanField(
+                default=False, verbose_name="This invoice is payed.", editable=False
```
```suggestion
                default=False, verbose_name="This invoice has been paid.", editable=False
```

towo commented on devroom/migrations/0001_initial.py:
```diff
@@ -17,36 +17,122 @@ class Migration(migrations.Migration):
+                ),
+                (
+                    "homepage",
+                    models.URLField(blank=True, verbose_name="Project homepage url"),
```
```suggestion
                    models.URLField(blank=True, verbose_name="Project homepage URL"),
```

towo reviewed (changes requested):
Couple of language adjustments I noticed while doing a drive-by review.

garw committed:
Fix TokenAuthenticationBackend

Django changed the signature of authenticate() on the caller
end on and for the authentication backends to always include
the request. Adapt this and add the TokenAuthenticationBackend
to the default backends. Additional backends can be added by a local
configuration by just appending to AUTHENTICATION_BACKENDS.

garw committed:
Fix case typo for template loaders

If debug mode is off, a cached template loader is should be used.
However, due to wrong case, the option could not be applied, which lead
to a startup failure if not run in debug mode.

bernharddick committed:
Pin to Django 3

bernharddick committed:
Fix defective URL entry

Missing character has been added
